src/dashboard/tests/test_helpers.py

```diff
@@ -0,0 +1,159 @@
+# -*- coding: utf-8 -*-
+
+from __future__ import unicode_literals
+
+import __builtin__
+import json
+import os
+import requests
+import tempfile
+
+from amclient import AMClient
+from components import helpers
+
+
+def test_stream_mets_from_disconnected_storage(mocker):
+    custom_error_message = "Error connecting to the storage service"
+    mocker.patch.object(AMClient, "__init__", return_value=None)
+    mocker.patch.object(
+        helpers, "get_setting", return_value="http://storage-service-url/"
+    )
+    mocker.patch.object(
+        AMClient,
+        "extract_file",
+        side_effect=requests.exceptions.ConnectionError(custom_error_message),
+    )
+    response = helpers.stream_mets_from_storage_service(
+        transfer_name="mets_transfer", sip_uuid="11111111-1111-1111-1111-111111111111"
+    )
+    err = json.loads(response.content)
+    assert err.get("success") is False
+    assert custom_error_message in err.get("message")
+    assert response.status_code == 503
+
+
+def test_stream_mets_from_storage_no_file(mocker, tmp_path):
+    custom_error_message = "METS file cannot be opened"
+    mocker.patch.object(AMClient, "__init__", return_value=None)
+    mocker.patch.object(
+        helpers, "get_setting", return_value="http://storage-service-url/"
+    )
+    mocker.patch.object(AMClient, "extract_file")
```

Comment:
I think that if you don't care about the return value of `extract_file` here you could mock `AMClient` entirely.

Reply:
I tried, but I'm not sure it would work. what we know is that AMClient itself is a class object whose init doesn't return anything, and whose member functions perform the work. I tried `mocker.patch("amclient.AMClient")` is there another trick I'm missing maybe?

Reply:
You could mock the `AMClient` class from the `helpers` module (or anything else in that module) like this:
 
```python
def mock_amclient_details(mocker, side_effect=None):
    mocker.patch(
        "components.helpers.get_setting", return_value="http://storage-service-url/"
    )
    amclient_mock = mocker.Mock(**{"extract_file.side_effect": side_effect})
    mocker.patch("components.helpers.AMClient", return_value=amclient_mock)
````

Reply:
Oh that's quite nice. Thanks Douglas. I'll put this way in.